homeassistant/components/alarmdecoder/config_flow.py

```diff
@@ -101,8 +101,8 @@ async def async_step_protocol(self, user_input=None):
 
             controller = AdExt(device)
             try:
-                with controller:
-                    controller.open(baudrate=baud)
+                await self.hass.async_add_executor_job(controller.open, baud)
+                await self.hass.async_add_executor_job(controller.close)
```

Comment:
We can make a new function that wraps these two calls to the controller and then we just schedule that function once on the executor. That will be more effective since there's an overhead to the context switches.

Reply:
@MartinHjelmare, consolidated in https://github.com/home-assistant/core/pull/40037/commits/5d9484f90335241d50f22a5a380dccf7e68a4cb5 :+1: